packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -231,7 +244,7 @@ export class Badge {
           ["show"]: visible,
           ["hide"]: !visible,
         }}
-        id="badge"
+        id={badgeId}
```

Comment:
Maybe we should make badge id conditionally set here. Either have it be `this.id` (if developers set an id on the element) or null?